modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts

```diff
@@ -620,4 +623,28 @@ export abstract class BaseCoin implements IBaseCoin {
   setCoinSpecificFieldsInIntent(intent: PopulatedIntent, params: PrebuildTransactionWithIntentOptions): void {
     return;
   }
+
+  /** @inheritDoc */
+  assertIsValidKey(params: AuditKeyParams): void {
+    let decryptedKey: string;
+
+    try {
+      decryptedKey = sjcl.decrypt(params.walletPassphrase, params.encryptedPrv);
+    } catch (e) {
+      throw new Error(`failed to decrypt prv: ${e.message}`);
+    }
+    this.auditDecryptedKey({ ...params, prv: decryptedKey });
+  }
+
+  /**
+   * Audit if a decrypted key is valid.
+   * @param {AuditDecryptedKeyParams} params - parameters for auditing the decrypted key
+   * @param {string} params.prv - the decrypted private key
+   * @param {string} params.publicKey - the public key, or common keychain
+   * @param {string} params.multiSigType - the multi-sig type, if applicable
+   * @returns {AuditKeyResponse} - result of the audit
+   */
+  protected auditDecryptedKey(params: AuditDecryptedKeyParams) {
+    throw new NotImplementedError('auditDecryptedKey is not implemented for this coin');
+  }
```

Comment:
```suggestion
  abstract auditDecryptedKey(params: AuditDecryptedKeyParams);
```

that way you enforce that subclasses implement it at compile time